shell/platform/darwin/macos/framework/Headers/FlutterViewController.h

```diff
@@ -110,6 +111,39 @@ FLUTTER_DARWIN_EXPORT
                                nibName:(nullable NSString*)nibName
                                 bundle:(nullable NSBundle*)nibBundle NS_DESIGNATED_INITIALIZER;
 
+/**
+ * The identifier for this view controller, if it is attached.
+ *
+ * The identifier is assigned when the view controller is attached to a
+ * `FlutterEngine. See `FlutterEngine#addViewController:` for details.
+ *
+ * If the view controller is detached (see `FlutterViewController#attached`),
+ * reading this property throws an assertion.
+ */
+@property(nonatomic, readonly) FlutterViewId viewIdentifier;
+
+/**
+ * Whether this view controller corresponds to the implicit view, if it is
+ * attached.
+ *
+ * If the view controller is detached (see `FlutterViewController#attached`),
+ * reading this property throws an assertion.
+ *
+ * The implicit view is a special view for backwards compatibility.
+ * The engine assumes it can always render to this view, even if the app has
+ * destroyed the window for this view.
+ *
+ * Today, the implicit view is the first view that is created. It is the only
+ * view that can be created before the engine is launched.
+ *
+ * The embedder must ignore presents to this view before it is created and
+ * after it is destroyed.
+ *
+ * See:
+ * https://api.flutter.dev/flutter/dart-ui/PlatformDispatcher/implicitView.html
+ */
+- (BOOL)isImplicitView;
```

Comment:
I understand why we need to expose this in dart:ui to the framework, but why do we need this in the public app-facing interface?

Reply:
Removed. Thanks for pointing out.